event_name_factory.go

```diff
@@ -0,0 +1,58 @@
+/*
+ * Flow Go SDK
+ *
+ * Copyright 2019 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package flow
+
+import (
+	"fmt"
+)
+
+type eventNameFactory struct {
+	address      string
+	contractName string
+	eventName    string
+}
+
+func (f eventNameFactory) WithAddressString(address string) eventNameFactory {
+	f.address = address
+	return f
+}
+
+func (f eventNameFactory) WithAddress(address Address) eventNameFactory {
+	f.address = address.Hex()
+	return f
+}
+
+func (f eventNameFactory) WithContractName(contract string) eventNameFactory {
+	f.contractName = contract
+	return f
+}
+
+func (f eventNameFactory) WithEventName(event string) eventNameFactory {
+	f.eventName = event
+	return f
+}
+
+func (f eventNameFactory) Build() string {
+	return fmt.Sprintf("A.%s.%s.%s", f.address, f.contractName, f.eventName)
+}
+
+// NewEvent helper function for constructing event names
+func NewEvent() eventNameFactory {
```

Comment:
```suggestion
func NewEventTypeFactory() eventTypeFactory {
```
Also `eventNameFactory` -> `eventTypeFactory` (you can also move this code into `event.go`). 

I suggest making the name more explicit to keep in line with cadence terminology and avoid confusion with `cadence.Event`. Go SDK already calls this `Type` in `event.go`. The general `Event` type is expected to have fields and values.

https://github.com/onflow/cadence/blob/52549529e96b6094bb726e5e50d63f34e2eb292b/types.go#L1595-L1601

https://github.com/onflow/flow-go-sdk/blob/3f7414701e1aab43b272f791b7385ae9e329aca8/event.go#L42

cc @sideninja it was part of the original design would like to know your thoughts as well

Reply:
@nvdtf  amended here https://github.com/onflow/flow-go-sdk/pull/449/commits/155a11e76499d84ef5487b8d1ca7cd29d57cf428

Reply:
Thanks! Some leftovers:
* filename`event_name_factory.go` should be changed. I suggest moving this code into `events.go` or if you want to keep it separate, rename the file to `event_type_factory.go`
* `event_name_factory_test.go` -> `event_type_factory_test.go`
* Test function name ` TestEventNameFactory` -> ` TestEventTypeFactory`

Reply:
@nvdtf fixed here https://github.com/onflow/flow-go-sdk/pull/449/commits/f99825351b3ebea44e6d460369939f32c572819c.

Thank you very much for pointing out.